app/oauth2/oauth2-route.js

```diff
@@ -1,17 +1,27 @@
 const accessTokenRequest = require('./access-token-request');
+const hasAnAttribute = require('../util/class-attributes-helper');
+
 const config = require('config');
 const COOKIE_ACCESS_TOKEN = 'accessToken';
 
 const oauth2Route = (req, res, next) => {
   accessTokenRequest(req)
     .then(result => {
-      res.cookie(COOKIE_ACCESS_TOKEN, result.access_token,
-        {
-          maxAge: result.expires_in * 1000,
-          httpOnly: true,
-          secure: config.get('security.secure_auth_cookie_enabled')
+      if( hasAnAttribute(result,'access_token')) {
+        res.cookie(COOKIE_ACCESS_TOKEN, result.access_token,
+          {
+            maxAge: result.expires_in * 1000,
+            httpOnly: true,
+            secure: config.get('security.secure_auth_cookie_enabled')
+          });
+        res.status(204).send();
+      } else {
+        next({
```

Comment:
The change here could be a breaking change for some services? Some tests may rely on a previous response structures. Just to make sure we coordinate, if this is the case.

Reply:
Yes I have been doing some end to end testing. but yes  We need to coordinate it. I might need some advices.